### PR TITLE
Add merge_mode join

### PR DESCRIPTION
**Pull request by EderSantana**
Some advanced models are made easier with this simple addition. It does not break anything, just add an extra option to Merge layers. This option allows it to work with layers that get several inputs separately.
For example, I'm writing a DRAW model using Keras and I believe this is the only non-standard thing that the model would require to run with Keras out of the box.

### CODE REVIEW

**Comment by fchollet** [comment]
I'm not against it, but this breaks the Keras API conventions. Multi-output layers are supposed to return a dictionary mapping a string names to tensors, not a list of tensors...

**Comment by EderSantana** [comment]
Ok! I think merge join can return a dictionary with very little further work. Do you think that would be good? I can update the PR

**Comment by fchollet** [comment]
Yes, that sounds good!

**Comment by EderSantana** [comment]
See how it looks now. Let me know which further changes should I do here.

File: keras/layers/core.py [hunk 1]
```diff
@@ -196,6 +198,11 @@ def get_output(self, train=False):
         elif self.mode == 'concat':
             inputs = [self.layers[i].get_output(train) for i in range(len(self.layers))]
             return T.concatenate(inputs, axis=self.concat_axis)
+        elif self.mode == 'join':
+            inputs = OrderedDict()
+            for i in range(len(self.layers)):
+                inputs[self.layers[i].get_output(train).name] = self.layers[i].get_output(train)
```
Comment (fchollet) [inline keras/layers/core.py]: What if the layer does not have a name, though? There should be an exception or a fallback (preferably an exception, for maximum explicitness).

Comment (EderSantana) [inline keras/layers/core.py]: Agree! I'll update the code.

**Comment by EderSantana** [comment]
I'm raising an exception if using unnamed inputs. Check if you agree with the message.

**Comment by fchollet** [comment]
Cool, looks good!

**Pull request merged**